File: java/PE037.java
public class PE037 {
    public static void main(String[] args) {
        int sum = 0;
        int count = 0;
        for(int i = 11; count < 11; i++) {
            if(isPrime(i)) {
                if(isLeft(i) && isRight(i)) {
                    count++;
                    sum += i;
                }
            }
        }
        System.out.println(sum);
    }

    private static boolean isLeft(int n)
    {
        while(n > 0) {
            int i;
            for(i = 1; i <= n; i *= 10) ;
            i /= 10;
            n = n - (n/i)*i;
            if(!isPrime(n))
                return false;
        }
        return true;
    }

    private static boolean isRight(int n)
    {
        while(n != 0) {
            n /= 10;
            if(!isPrime(n))
                return false;
        }
        return true;
    }
    private static boolean isPrime(int n)
    {
        if(n == 1)
            return false;
        int k = (int)Math.sqrt(n);
        for(int i = 2; i <= k; i++) {
            if(n%i == 0)
                return false;
        }
        return true;
    }
}

File: cpp/PE037.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool isPrime(int n)
{
    if(n == 1)
        return false;
    else if(n < 4)
        return true;
    else if(n%2 == 0)
        return false;
    else if(n < 9)
        return true;
    else if(n%3 == 0)
        return false;
    else{
        int r = (int)sqrt((double)n);
        int f = 5;
        while(f <= r){
            if(n%f == 0)
                return false;
            else if(n%(f+2) == 0)
                return false;
            f += 6;
        }
    }
    return true;
}
bool isTruncatable(int n){
    int m = n, i = 1;
    m /= 10;
    while(m){
        if(!isPrime(m))
            return false;
        i *= 10;
        m /= 10;
    }
    m = n;
    while(i != 1){
        m = m%i;
        if(!isPrime(m))
            return false;
        i /= 10;
    }
    return true;
}
int main()
{
    int i, count = 0, sum = 0;
    for(i = 11; ; i += 2){
        if(isPrime(i) && isTruncatable(i)){
            count++;
            sum += i;
        }
        if(count == 11)
            break;
    }
    cout << sum << endl;
    return 0;
}
